antenna/ext/sqs/crashpublish.py

```diff
@@ -0,0 +1,169 @@
+        doc="AWS region to connect to. For example, ``us-west-2``",
+    )
+    required_config.add_option(
+        "endpoint_url",
+        default="",
+        alternate_keys=["root:s3_endpoint_url"],
+        doc=(
+            "endpoint_url to connect to; None if you are connecting to AWS. For "
+            "example, ``http://localhost:4569/``."
+        ),
+    )
+    required_config.add_option("queue_name", doc="The AWS SQS queue name.")
+
+    def __init__(self, config):
+        super().__init__(config)
+
+        self.queue_name = self.config("queue_name")
+        self.client = self._build_client()
+        self.queue_url = self.client.get_queue_url(QueueName=self.queue_name)[
+            "QueueUrl"
+        ]
+
+        register_for_verification(self.verify_queue)
+
+    @retry(
+        retryable_exceptions=[
+            # FIXME(willkg): Seems like botocore always raises ClientError
+            # which is unhelpful for granularity purposes.
+            ClientError,
+            # This raises a ValueError "invalid endpoint" if it has problems
+            # getting the s3 credentials and then tries "s3..amazonaws.com"--we
+            # want to retry that, too.
+            ValueError,
+        ],
+        wait_time_generator=wait_times_connect,
+        sleep_function=gevent.sleep,
+        module_logger=logger,
+    )
+    def _build_client(self):
+        # Either they provided ACCESS_KEY and SECRET_ACCESS_KEY in which case
+        # we use those, or they didn't in which case boto3 pulls credentials
+        # from one of a myriad of other places.
+        # http://boto3.readthedocs.io/en/latest/guide/configuration.html#configuring-credentials
+        session_kwargs = {}
+        if self.config("access_key") and self.config("secret_access_key"):
+            session_kwargs["aws_access_key_id"] = self.config("access_key")
+            session_kwargs["aws_secret_access_key"] = self.config("secret_access_key")
+        session = boto3.session.Session(**session_kwargs)
+
+        kwargs = {
+            "service_name": "sqs",
+            "region_name": self.config("region"),
+        }
+        if self.config("endpoint_url"):
+            kwargs["endpoint_url"] = self.config("endpoint_url")
+
+        return session.client(**kwargs)
+
+    def verify_queue(self):
+        """Verify queue can be published to by publishing fake crash id."""
+        self.client.send_message(QueueUrl=self.queue_url, MessageBody="test")
+
+    def check_health(self, state):
+        """Check AWS SQS connection health."""
+        try:
+            self.client.get_queue_url(QueueName=self.queue_name)
+        except Exception as exc:
+            state.add_error("SQSCrashPublish", repr(exc))
+
+    def publish_crash(self, crash_report):
+        """Publish a crash id to an AWS SQS queue."""
+        crash_id = crash_report.crash_id
+        self.client.send_message(QueueUrl=self.queue_url, MessageBody=crash_id)
```

Comment:
The message body can be unformatted text, xml, or JSON.

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sqs.html#SQS.Client.send_message

This is different than pubsub where we had to send bytes.